### DISCUSSION TITLE
Get RAW request for logging/debugging purposes.

### DISCUSSION THREAD

**Discussion by eNTi** [canton7/RestEase#206] [Q&A] [answered]
I need to check what exact string that is sent and received over the wire before serialization / after deserialization. Any way to do that?

**Comment by canton7** [comment, marked as answer]
When you say "before serialization/deserialization", do you mean after serialization and before deserialization, or before serialization and after deserialization?

**Reply by canton7** [reply]
Also see [this doc](https://github.com/canton7/RestEase/blob/a5a6514f54d7e41a5ed1acfcf6d46493402d8cea/README.md#custom-httpclient), which has an example of a custom `HttpClientHandler` which logs all requests.

**Reply by eNTi** [reply]
Sorry, i meant before serialization and after deserialization. I need the string representation not the bytes.

**Reply by canton7** [reply]
You can read the body's bytes as a string, see that link.

When you say the "string representation", what exactly are you referring to? The serialized body? The URI, including query parameters?

If you're referring to the body, what if it's not a string? What if it's a `Stream` or `byte[]` which gets serialized directly to bytes, or perhaps a `HttpContent` which could be anything? You could hook into the `RequestBodySerializer` (see your other question) which will let you log every time a body is serialized, and you can get some other information on the request, but that won't get called if there is no body.

**Comment by eNTi** [comment]
I mean the header as well as the json content in the body of a POST request. If you know SoapUI for example you can see "Raw Requests", which will show you the header + body.
[SoapUI](https://i.imgur.com/p9mlxEr.png)

**Reply by canton7** [reply]
The easiest way would be to use a custom `HttpClientHandler` and read the body as a string, see my first message.

**Reply by eNTi** [reply]
Thank you. I'll try that.
